traversal/selector/matcher_util.go

```diff
@@ -0,0 +1,15 @@
+package selector
+
+import "io"
+
+type readerat struct {
+	io.ReadSeeker
+}
+
+func (r readerat) ReadAt(p []byte, off int64) (n int, err error) {
+	_, err = r.Seek(off, 0)
```

Comment:
nit: as an optimization, if you're already at the right offset, you could skip the seek. fine to leave for later, but I'd still keep a TODO, because this will matter for e.g. SectionReader - one vs two calls per read.

Reply:
leaving as todo, because i think there's a tradeoff between either calling seek to get current offset each time, or keeping extra state to track.

Reply:
Keeping extra state seems easy enough, given that you need a mutex anyway :) but fine to leave as a TODO for sure